src/com/sun/jna/Memory.java

```diff
@@ -85,6 +87,10 @@ public SharedMemory(long offset, long size) {
             this.size = size;
             this.peer = Memory.this.peer + offset;
         }
+        @Override
+        protected void finalize() {
+            dispose();
+        }
```

Comment:
Please don't introduce another override of `Object#finalize`. I tried to reduce the deprecation warnings and at some point in time finalizers will be removed (maybe we are lucky and project panama will have made JNA redundant by than, but ...)

Reply:
okay, i will keep this in mind and will rework this change.